Sources/MapboxCoreNavigation/FeedbackEvent.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+/**
+ Feedback event that can be created using `NavigationEventsManager.createFeedback()`.
+ Use `NavigationEventsManager.sendFeedback(_:type:description:)` to send it to the server.
+ Conforms to the `Codable` protocol, so the application can store the event persistently.
+ */
+public class FeedbackEvent: Codable {
+    let coreEvent: CoreFeedbackEvent
+    
+    init(eventDetails: NavigationEventDetails) {
+        let dictionary = try? eventDetails.asDictionary()
+        coreEvent = CoreFeedbackEvent(timestamp: Date(), eventDictionary: dictionary ?? [:])
+    }
+    
+    func update(type: FeedbackType, source: FeedbackSource, description: String?) {
+        coreEvent.eventDictionary["feedbackType"] = type.description
+
+        // if there is a subtype for this event then append the subtype description to our list for this type of feedback
+        if let subtypeDescription = type.subtypeDescription {
+            var subtypeList = [String]()
+            if let existingSubtypeList = coreEvent.eventDictionary["feedbackSubType"] as? [String] {
```

Comment:
Should we silently fail here if `feedbackSubType` isn't an array of strings?

Reply:
We are not failing in this case, this logic creates an empty array in case we don't have an array already.

Reply:
Well, ignoring the value is kind of "silent fail". The code which doesn't do what it was asked for (even wrongly) is just a form of fail.